uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-fs/src/main/java/org/uberfire/java/nio/fs/file/SimpleFileSystemProvider.java

```diff
@@ -744,6 +622,118 @@ private GeneralPathImpl toGeneralPathImpl(final Path path) {
+        if (path.toFile().exists()) {
+            throw new FileAlreadyExistsException(path.toString());
+        }
+    }
+
+    private final class SimpleFileSystemDirectoryStream implements DirectoryStream<Path> {
+
+        private final class SimpleFileSystemIterator implements Iterator<Path> {
+
+            public boolean atEof = false;
+            private int i = -1;
+            private Path nextEntry = null;
+
+            @Override
+            public boolean hasNext() {
+                if (isClosed) {
+                    throw new IOException("This stream is closed.");
+                }
+                if (nextEntry == null && !atEof) {
+                    nextEntry = readNextEntry();
+                }
+                return nextEntry != null;
+            }
+
+            @Override
+            public Path next() {
+                final Path result;
+                if (isClosed) {
+                    throw new IOException("This stream is closed.");
+                }
+                if (nextEntry == null && !atEof) {
+                    result = readNextEntry();
+                } else {
+                    result = nextEntry;
+                    nextEntry = null;
+                }
+                if (result == null) {
+                    throw new NoSuchElementException();
+                }
+                return result;
+            }
+
+            private Path readNextEntry() {
+                if (atEof) {
+                    return null;
+                }
+
+                Path result = null;
+                while (true) {
+                    i++;
+                    if (i >= content.length) {
+                        atEof = true;
+                        break;
+                    }
+
+                    final Path path = content[i];
+                    if (filter.accept(path)) {
+                        result = path;
+                        break;
+                    }
+                }
+
+                return result;
+            }
+
+            @Override
+            public void remove() {
+                throw new UnsupportedOperationException();
+            }
+        }
+
+        private final Filter<Path> filter;
+        private final Path[] content;
+        boolean isClosed = false;
+
+        private SimpleFileSystemDirectoryStream(Filter<Path> filter, Path[] content) {
+            this.filter = filter;
+            this.content = content;
+        }
+
+        @Override
+        public void close() throws IOException {
+            if (isClosed) {
+                throw new IOException("This stream is closed.");
+            }
+            isClosed = true;
+        }
+
+        @Override
+        public Iterator<Path> iterator() {
+            if (isClosed) {
+                throw new IOException("This stream is closed.");
+            }
+            return new SimpleFileSystemIterator();
+        }
+    }
+
     public enum OSType {
```

Comment:
i'd prefer to preserve the original method signatures, what would also help to visualize only the relevant changes

Reply:
It's SonarCloud rule violation again. Using anonymous Class, especially nested anonymous Class is discouraged. As a result, refactored them into inner Classes. It literally has no code change to the original implementation.

Reply:
ignore sonarCloud :)

Reply:
Reverted back to original style, but SonarCloud might not be happy :(